[Python] string formatting concept exercise (#3034)
exercism/v3: languages/python/concepts/string-formatting/about.md

## String Formatting in Python

> 

Reviewer: ```suggestion
The [Zen of Python][zen-of-python] asserts there should be “one _obvious_ way to do something in Python".  But when it comes to string formatting, things are a little .... _less zen_.   It can be surprising to find out that there are **four** major ways to perform string formatting in Python - each for a different scenario.
```

```python
>> from string import Template
>> template_string = Template("The snake called `$snake_name` has escaped!")
>> template_string.substitute(snake_name=name)
'The snake called `Anna-conda` has escaped'
```

More information about `Template` string can be found in the Python [documentation][template-string].

## How to choose which formatting method to use?

1. The `f-string` is the newest and easiest to read, it should be preferred when using Python 3.6+.
2. Then `.format()` is versatile, very powerful and compatible with most versions of Python.
3. `Template string` can be used to mitigate risks when inputs from users need to be handled, for simplicity or multiple substitutions.
4. The `%` operator is not supported in some of the new distributions of Python, it is mostly used for compatibility with old code. this operator can lead to issues displaying non-ascii characters and it is less powerful than other methods.

If you want to go further: [all about formatting][all-about-formatting].

> 

Reviewer: ```suggestion

[zen-of-python]: https://www.python.org/dev/peps/pep-0020/
```

SyntaxError: f-string expression part cannot include '#'
```

> f-string cannot be used together with the GNU gettext API for internationalization (I18N) and localization (L10N), `str.format()` needs to be used instead. Using an expression inside the brackets `{}` is similar to use `eval()` or `exec()`, so it isn't very safe.

Reviewer: ```suggestion
String interpolation cannot be used together with the GNU gettext API for internationalization (I18N) and localization (L10N), so `str.format()` needs to be used instead of an `f-string` in translation scenarios. Keep in mind that using an expression inside the brackets `{}` is similar to using `eval()` or `exec()`, so it isn't very safe and should be used sparingly.
```
